controllers/overlaynetwork_controller.go

```diff
@@ -47,9 +53,18 @@ type OverlayNetworkReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *OverlayNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	overlayNetwork := &selfservicev1.OverlayNetwork{}
+	if err := r.Client.Get(ctx, req.NamespacedName, overlayNetwork); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get OverlayNetwork %q: %v", req.NamespacedName, err)
+	}
 
-	// TODO(user): your logic here
+	netAttachDef := renderNetAttachDef(overlayNetwork)
+	if err := r.Client.Create(ctx, netAttachDef); err != nil {
+		return ctrl.Result{}, err
```

Comment:
If a NAD with this CR ownerRef is there, it is fine. We shouldn't return an error.
If another NAD is there, we should return an error.